### PR TITLE
(#102) HTTP Status Code Test

### LINKED ISSUE #102: Migration of HTTP status code test
## As a user, I want correct status codes for page requests.

> **ESTIMATE** TBD

### Acceptance Criteria

```gherkin
 Scenario Outline: Check received status code from get request
        Given user is navigating to "<path>"
        Then correct status code "<code>" is received

        Examples:
            | path                                                                                                       | code |
            | /Common/PopUps/popDefinition.aspx?id=485395&version=healthprofessional&language=English&dictionary=genetic | 200  |
            | /news-events/cancer-currents-blog                                                                          | 200  |
            | /espanol/noticias/temas-y-relatos-blog                                                                     | 200  |
            | /types/breast/research                                                                                     | 200  |
            | /espanol/tipos/seno/investig                                                                               | 200  |
            | /types/breast                                                                                              | 200  |
            | /espanol/tipos/seno                                                                                        | 200  |
            | /types/breast/hp                                                                                           | 200  |
            | /about-cancer/coping/feelings/relaxation/2019-investigators-site                                           | 200  |
            | /about-cancer                                                                                              | 200  |
            | /espanol/cancer                                                                                            | 200  |
            | /news-events/press-releases/2018                                                                           | 200  |
            | /espanol/noticias/comunicados-de-prensa/2018                                                               | 200  |
            | /about-nci/organization/screen-to-save-infographic                                                         | 200  |
            | /espanol/infografia-nci                                                                                    | 200  |
            | /research/progress/discovery/gutcheck-intro-video                                                          | 200  |
            | /nano                                                                                                      | 200  |
            | /nano/espanol/cancer/sobrellevar/sentimientos                                                              | 200  |
            | /pediatric-adult-rare-tumor                                                                                | 200  |
            | /pediatric-adult-rare-tumor/espanol                                                                        | 200  |
            | /espanol/tipos/rinon                                                                                       | 404  |
            | /grants-training/apply-grant/peer-review                                                                   | 404  |
            | /about-cancer/coping/feelings/relaxation/chicken                                                           | 404  |
            | /nano/abcdef                                                                                               | 404  |
            | /nano/espanol/cancer/sobrellevar/sentim                                                                    | 404  |

```

### Resources:
* Feature: JIRA Ticket #
* Mockups: *Here goes a URL to or the name of the mockup(s) in inVision*
* Web Page: *Here goes a URL to an existing web page*

### Notes

*Some complementary notes if necessary:*

* > Here goes a quote from an email
* Here goes whatever useful information can exist…

---

### Prerequisites
- *required dependency for identifying the problem, scope and then solution for this issue*
- [ ] Short Spike Description - Issue #9999

### Tasks
- [ ] Short Task Description - Issue #9999
- [ ] Short Task Description - Issue #9999

### Technical Notes
- More notes, or technical nuances/direction

### PR DESCRIPTION
**Pull request by SupriyaSaxena0605**
Added feature file.
Added tests
Closes #102

### CODE REVIEW

**Comment by SupriyaSaxena0605** [comment]
@KateMashkinaNIH Please review
image

File: cypress/integration/Crosscutting/HttpStatusCodeTest.feature [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+            | /news-events/press-releases/2018                                                                           | 200  |
+            | /espanol/noticias/comunicados-de-prensa/2018                                                               | 200  |
+            | /about-nci/organization/screen-to-save-infographic                                                         | 200  |
+            | /espanol/infografia-nci                                                                                    | 200  |
+            | /research/progress/discovery/gutcheck-intro-video                                                          | 200  |
+            | /nano                                                                                                      | 200  |
+            | /nano/espanol/cancer/sobrellevar/sentimientos                                                              | 200  |
+            | /pediatric-adult-rare-tumor                                                                                | 200  |
+            | /pediatric-adult-rare-tumor/espanol                                                                        | 200  |,
```
Comment (KateMashkinaNIH) [inline cypress/integration/Crosscutting/HttpStatusCodeTest.feature:26]: please add more paths here

```
  | /publications/dictionaries/cancer-terms/def/45413                | 200  |
      | /publications/dictionaries/cancer-drug                           | 200  |
      | /espanol/publicaciones/diccionario/ampliar/C                     | 200  |
      | /about-cancer/treatment/clinical-trials/search                   | 200  |
      | /about-cancer/treatment/clinical-trials/search/advanced          | 200  |
      | /research/resources                                              | 200  |
      | /research/resources/search?from=0&researchAreas=cancer_biology   | 200  |
      | /search/results?swKeyword=cancer                                 | 200  |
```

File: cypress/integration/Crosscutting/HttpStatusCodeTest.feature [hunk 2]
```diff
@@ -0,0 +1,26 @@
+Feature: Test for checking HTTP Status Code
+
+  Scenario Outline: Check received status code from get request
+        Given user is navigating to "<path>"
+        Then correct status code "<code>" is received for the "<path>"
```
Comment (KateMashkinaNIH) [inline cypress/integration/Crosscutting/HttpStatusCodeTest.feature:5]: <code> should be an int, not a string

File: cypress/integration/Crosscutting/HttpStatusCodeTest/index.js [hunk 3]
```diff
@@ -0,0 +1,11 @@
+// <reference types="Cypress" />
+import { Given, And, Then } from 'cypress-cucumber-preprocessor/steps';
+
+Then('correct status code {string} is received for the {string}', (code, path) => {
```
Comment (KateMashkinaNIH) [inline cypress/integration/Crosscutting/HttpStatusCodeTest/index.js:4]: dont forget to change codes' type from {string} to {int}

File: cypress/integration/Crosscutting/HttpStatusCodeTest/index.js [hunk 4]
```diff
@@ -0,0 +1,11 @@
...
+    cy.request({
+        url: path,
+        followRedirect: false,
```
Comment (KateMashkinaNIH) [inline cypress/integration/Crosscutting/HttpStatusCodeTest/index.js:7]: this should not be here, remove followRedirect arg

File: cypress/integration/Crosscutting/HttpStatusCodeTest/index.js [hunk 5]
```diff
@@ -0,0 +1,11 @@
+// <reference types="Cypress" />
+import { Given, And, Then } from 'cypress-cucumber-preprocessor/steps';
+
+Then('correct status code {string} is received for the {string}', (code, path) => {
+    cy.request({
+        url: path,
+        followRedirect: false, 
+      }).then((resp) => {
+        expect(resp.status).toString().match(code);      })
```
Comment (KateMashkinaNIH) [inline cypress/integration/Crosscutting/HttpStatusCodeTest/index.js:9]: this should be equal
` expect(resp.status).to.eq(code);`

File: cypress/integration/Crosscutting/HttpStatusCodeTest.feature [hunk 6]
```diff
@@ -0,0 +1,26 @@
+Feature: Test for checking HTTP Status Code
+
+  Scenario Outline: Check received status code from get request
+        Given user is navigating to "<path>"
```
Comment (KateMashkinaNIH) [inline cypress/integration/Crosscutting/HttpStatusCodeTest.feature:4]: no need to navigate at all, you can delete this step , sending request in the second step is more than enough

**Review by KateMashkinaNIH — CHANGES_REQUESTED** [review]

**Comment by SupriyaSaxena0605** [comment]
image

@KateMashkinaNIH Please review.

File: cypress/integration/Crosscutting/HttpStatusCodeTest.feature (hunk 6 above)
Comment (SupriyaSaxena0605) [reply cypress/integration/Crosscutting/HttpStatusCodeTest.feature]: done

File: cypress/integration/Crosscutting/HttpStatusCodeTest/index.js (hunk 5 above)
Comment (SupriyaSaxena0605) [reply cypress/integration/Crosscutting/HttpStatusCodeTest/index.js]: done

File: cypress/integration/Crosscutting/HttpStatusCodeTest/index.js (hunk 4 above)
Comment (SupriyaSaxena0605) [reply cypress/integration/Crosscutting/HttpStatusCodeTest/index.js]: done

File: cypress/integration/Crosscutting/HttpStatusCodeTest/index.js (hunk 3 above)
Comment (SupriyaSaxena0605) [reply cypress/integration/Crosscutting/HttpStatusCodeTest/index.js]: done

File: cypress/integration/Crosscutting/HttpStatusCodeTest.feature (hunk 2 above)
Comment (SupriyaSaxena0605) [reply cypress/integration/Crosscutting/HttpStatusCodeTest.feature]: done

File: cypress/integration/Crosscutting/HttpStatusCodeTest.feature (hunk 1 above)
Comment (SupriyaSaxena0605) [reply cypress/integration/Crosscutting/HttpStatusCodeTest.feature]: done

**Review by kate-mashkina — APPROVED** [review]

**Review by KateMashkinaNIH — APPROVED** [review]

**Pull request merged**